ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -549,6 +549,96 @@ private static void createPhaseTapChanger(UcteAngleRegulation ucteAngleRegulatio
+                .setLowTapPosition(lowerRatioTap)
+                .setTapPosition(uctePhaseRegulation.getNp())
+                .setLoadTapChangingCapabilities(!Double.isNaN(uctePhaseRegulation.getU()));
+        if (!Double.isNaN(uctePhaseRegulation.getU())) {
+            rtca.setLoadTapChangingCapabilities(true)
+                    .setRegulating(true)
+                    .setTargetV(uctePhaseRegulation.getU())
+                    .setTargetDeadband(0.0)
+                    .setRegulationTerminal(transformer.getTerminal1());
+        }
+        double rhoInit = 1.;
+        double nrduInit = 0.;
+        for (int i = lowerRatioTap; i <= Math.abs(lowerRatioTap); i++) {
+            double rho = 1 / (1 + i * uctePhaseRegulation.getDu() / 100);
+            rtca.beginStep()
+                    .setRho(rho)
+                    .setR(0)
+                    .setX(0)
+                    .setG(0)
+                    .setB(0)
+                    .endStep();
+            if (i == uctePhaseRegulation.getNp()) {
+                rhoInit = rho;
+                nrduInit = i * uctePhaseRegulation.getDu() / 100;
+            }
+        }
+        rtca.add();
+
+        int lowerPhaseTap = getLowTapPosition(ucteAngleRegulation, transformer);
+        PhaseTapChangerAdder ptca = transformer.newPhaseTapChanger()
+                .setLowTapPosition(lowerPhaseTap)
+                .setTapPosition(ucteAngleRegulation.getNp())
+                .setRegulationValue(ucteAngleRegulation.getP())
+                .setRegulationMode(PhaseTapChanger.RegulationMode.FIXED_TAP)
+                .setRegulating(false);
+
+        if (!Double.isNaN(ucteAngleRegulation.getP())) {
+            ptca.setRegulationMode(PhaseTapChanger.RegulationMode.ACTIVE_POWER_CONTROL)
+                    .setTargetDeadband(0.0)
+                    .setRegulationTerminal(transformer.getTerminal1())
+                    .setRegulating(false); // should be set to true but many divergence on files are observed.
+        }
+
+        for (int i = lowerPhaseTap; i <= Math.abs(lowerPhaseTap); i++) {
+            double rho;
+            double alpha;
+            double dx = i * ucteAngleRegulation.getDu() / 100 * Math.cos(Math.toRadians(ucteAngleRegulation.getTheta()));
+            double dy = i * ucteAngleRegulation.getDu() / 100 * Math.sin(Math.toRadians(ucteAngleRegulation.getTheta()));
+            double dyEq = dy * (1. + nrduInit);
+            double dxEq = dx * (1. + nrduInit) + nrduInit;
+            switch (ucteAngleRegulation.getType()) {
+                case ASYM:
+                    rho = 1d / Math.hypot(dyEq, 1d + dxEq) / rhoInit; // the formula = 1d / Math.hypot(dyEq, 1d + dxEq) already takes into account rhoInit, so we divide by rhoInit that will be carried by the ratio tap changer
+                    alpha = Math.toDegrees(Math.atan2(dy, 1 + dx));
+                    break;
+
+                case SYMM:
```

Comment:
@phiedw here in case of symmetrical phase tap changer, no difference with classical treatment.

Reply:
I checked with a Fresnel diagram and the composition of ratio + angle taps should only change the global ratio which is carried by the ratio tap changer. As I understand the Fresnel Diagram, this doesn't change the alpha.
The diffrence with the asymmetrical phase shifter is that the formula is true for all ratio taps for the symmetrical one.

Reply:
Sorry indeed, I didn't understand that your work was not finished. So I will wait for symmetrical support to perform testing. As you see, I have simplified your equations to use only the current rho of the ratio tap changer. But I have to explain that in your documentation. Just tell me if it is okay for you.

Reply:
yes it is OK